### PR TITLE
Allow manual setting of hostname

### PR DESCRIPTION
**Pull request by gpiper14**
Implement #814
### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update documentation
- [x] Update [CHANGELOG.md](CHANGELOG.md)
- [x] Update [supported-technologies.asciidoc](docs/supported-technologies.asciidoc)
- [x] Added an API method or config option? Document in which version this will be introduced.

### CODE REVIEW

**Comment by gpiper14** [comment]
@felixbarny ready for a review

File: apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java [hunk 1]
```diff
@@ -114,6 +114,12 @@
             "the recommended value for this field is the commit identifier of the deployed revision, " +
             "e.g. the output of git rev-parse HEAD.")
         .build();
+    
+    private final ConfigurationOption<String> hostname = ConfigurationOption.stringOption()
+        .key("hostname")
+        .configurationCategory(CORE_CATEGORY)
+        .description("Allows for the reported hostname to be manually specified.  If unset the hostname will be looked up.")
```
Comment (felixbarny) [inline apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java]:
```suggestion
        .description("Allows for the reported hostname to be manually specified. If unset the hostname will be looked up.")
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # Next
 
 ## Features
-
+ * Add ability to manually specify reported hostname
```
Comment (felixbarny) [inline CHANGELOG.md]:
```suggestion
 * Add ability to manually specify reported [hostname](https://www.elastic.co/guide/en/apm/agent/java/current/config-core.html#config-hostname)
```

**Comment by felixbarny** [comment]
Jenkins run the tests please

**Comment by gpiper14** [comment]
> 
> 
> Thanks for the PR ❤️
> 
> Could you execute the `ConfigurationExporterTest`, so that the generated documentation updates?

Ok,  i'll try to get that up tonight.  My work computer seems to be holding a grudge against java 10+ and wont build :expressionless:

**Pull request merged**